---
title: IL23 Mini DMS Assay
output: github_document
---

1. [Barcode Sequencing Distributions](#part1)
2. [Inference and Stop Codon Effects](#part2)
3. [Visualizations](#part3)
4. [Contrasts](#part4)

```{r, echo=FALSE, warning=FALSE, message=FALSE}
knitr::opts_chunk$set(
  fig.path = "IL23_min_figs/",
  echo = FALSE,
  warning = FALSE,
  message = FALSE,
  dev = c("png", "pdf")
)
library(magrittr)
library(data.table)
library(ggrepel)
library(ggpubr)
library(vroom)
library(scales)
library(scico)
library(knitr)
library(ggbeeswarm)
library(fs)
library(patchwork)
library(ggnewscale)
library(ggrepel)
library(tidyverse)

source("../src/dms-analysis-utils.R")
#source("src/dms-analysis-utils.R")
```

```{r}
prefix <- "240222_VH00964_46_AAALYC7HV"
wt_aa <- read_tsv("../data/tyk2/tyk2-wt-aa.tsv")
```

## Barcode Sequencing Distributions <a name="part1"></a>

```{r, cache=TRUE, cache.lazy=FALSE}
mapped_counts <- data.table::fread(str_c("../pipeline/240222_VH00964_46_AAALYC7HV/16_tot_new.mapped-counts.tsv")) %>%
    rename("lib" = "DL4_1") %>%
    separate(oligo, c("lib", "chunk", "wt_aa", "pos",
                        "mut_aa", "wt_codon", "mut_codon"), "_") %>%
    mutate(condition_conc = as.factor(condition_conc),
        condition = as.factor(paste0(condition, condition_conc))) %>%
    group_by(sample) %>%
    mutate(mean_count = mean(log(as.numeric(count))),
        mut_aa = if_else(wt_aa == mut_aa | is.na(mut_aa), "WT", mut_aa),
        mut_aa = relevel(as.factor(mut_aa), ref = "WT"))

bc_counts_aa <- mapped_counts %>%
    filter(mut_aa != "WT") %>%
    group_by(sample,wt_aa,mut_aa,pos) %>%
    count()

bc_counts_codon <- mapped_counts %>%
    filter(mut_aa != "WT") %>%
    group_by(sample,wt_codon,mut_codon,pos) %>%
    count()

res_count <- bc_counts_aa %>%
    ggplot() +
        geom_boxplot(aes(x = sample, y = n)) +
        theme_pubr() +
        geom_hline(yintercept = 30, color = "red") +
        ylab("number of unique barcodes") + ggtitle("AA-level unique barcode counts")

codon_count <- bc_counts_codon %>%
    ggplot() +
        geom_boxplot(aes(x = sample, y = n)) +
        theme_pubr() +
        geom_hline(yintercept = 30, color = "red")  +
        ylab("number of unique barcodes") + ggtitle("Codon-level unique barcode counts")

res_count_log <- bc_counts_aa %>%
    ggplot() +
        geom_boxplot(aes(x = sample, y = log10(n))) +
        theme_pubr() +
        geom_hline(yintercept = log10(30), color = "red") +
        ylab("number of unique barcodes") + ggtitle("AA-level unique barcode counts, log10")

codon_count_log <- bc_counts_codon %>%
    ggplot() +
        geom_boxplot(aes(x = sample, y = log10(n))) +
        theme_pubr() +
        geom_hline(yintercept = log10(30), color = "red")  +
        ylab("number of unique barcodes") + ggtitle("Codon-level unique barcode counts, log10")
```

```{r, fig.width=20, fig.height=20}
res_count + codon_count
res_count_log + codon_count_log 
```

```{r, echo=TRUE, message=TRUE}
bc_counts_aa %>% group_by(sample) %>% summarise(mean = mean(n)) %>% knitr::kable(digits = 3)
```


```{r, fig.width=20, fig.height=20}
coverage_plots <- map(.x = unique(bc_counts_aa$sample),
                      .f = ~plot_coverage(bc_counts_aa, wt_aa, ., "il-23-mini"))
```

```{r, fig.width=20, fig.height=20}
coverage_plots[[1]]
```

```{r, fig.width=20, fig.height=10}
mapped_counts %>%
    ungroup() %>%
    count(sample, chunk) %>%
    filter(!is.na(chunk), sample != "sample") %>%
    ggplot() +
        geom_bar(aes(x = chunk, y = n), width = 1, color = NA, fill = "black", stat = "identity") +
        theme_bw() +
        facet_wrap(~sample, ncol = 4) +
        xlab("Chunk") + ylab("Total Number of Unique Barcodes\nAcross All Variants")
```

```{r, fig.width=20, fig.height=6}
mapped_counts %>%
    ungroup() %>%
    count(sample) %>%
    filter(sample != "sample") %>%
    ggplot() +
        geom_bar(aes(x = sample, y = n), width = 1, color = NA, fill = "black", stat = "identity") +
        theme_bw(base_size = 20) +
        xlab("Sample") + ylab("Total Number of Unique Barcodes\nAcross All Variants")
```

## Inference and Stop Codon Effects <a name="part2"></a>

In the plots below, the black curve shows the Z-statistic distribution of all
non-Stop variants across the entire length of the chunk.
Each red distribution shows the Z-statistics of Stop variants for one chunk
only, resulting in each facet having five red distributions (one for each chunk)
and one black distribution. 

```{r}
sumstats <- read_tsv(str_c("../sumstats/16_tot_new-combined.sumstats.tsv"), col_types = c("ccnccccnnnnnccnc")) %>%
    filter(grepl("aa",term)) %>%
    separate(term, c("condition", "aa"), ":") %>%
    mutate(condition = gsub("condition", "", condition),
           condition = gsub("0|10", "", condition),
           group = if_else(aa %in% c("mut_aa*", "mut_aaX"), chunk, "Non-Stop"),
           aa = gsub("mut_aa", "", aa),
           aa = if_else(aa %in% c("*","X"), "*", aa)) %>%
    rename("lib" = "DL4_1")

aa_uniq <- unique(sumstats$aa)
pos_uniq <- 1:348
```

```{r, fig.width=10, fig.height=6}
sumstats %>%
    mutate(group = if_else(aa == "*", chunk, "Non-Stop")) %>% 
    filter(chunk != 5) %>%
    ggplot() +
        geom_density(aes(x = statistic, group = group,
                         color = group != "Non-Stop",
                         size = group != "Non-Stop"), alpha = 0.5) +
        facet_grid(condition ~ lib) +
        theme_bw(base_size = 15) +
        scale_color_manual(values = c("FALSE" = "black", "TRUE" = "red")) +
        scale_size_manual(values = c("FALSE" = 2, "TRUE" = 0.75)) +
        theme(legend.pos = "none") +
        xlab("Z-Statistic") + ylab("Density") +
        ggtitle("IL23 Chunk 16 Mini DMS") +
        coord_cartesian(xlim = c(-10,7))
```

### Visualizations <a name="part3"></a>

```{r}
heat_l2fc <- sumstats %>%
    mutate(aa = gsub("mut_aa", "", aa),
           aa = if_else(aa %in% c("*", "X"), "Stop", aa)) %>%
    ggplot() +
            geom_tile(aes(x = pos, y = fct_rev(aa), fill =log2FoldChange)) +
            scale_fill_scico_mid(palette = "vik", mid = 0, oob = squish) +
            theme_pubr(base_size = 16, x.text.angle = 45) +
            theme(legend.text=element_text(size=14),
                  legend.key.size = unit(1,"cm"),
                  axis.text.y = element_text(size = 14)) +
            ylab("") + xlab("TYK2 Amino Acid Position") +
            facet_grid(condition ~ lib)

heat_z <- sumstats %>%
    mutate(aa = gsub("mut_aa", "", aa),
           aa = if_else(aa %in% c("*", "X"), "Stop", aa)) %>%
    rename("Z-Statistic" = "statistic") %>%
    ggplot() +
            geom_tile(aes(x = pos, y = fct_rev(aa), fill =`Z-Statistic`)) +
            scale_fill_scico_mid( limits = c(-15,5), palette = "vik", mid = 0, oob = squish) +
            theme_pubr(base_size = 16, x.text.angle = 45) +
            theme(legend.text=element_text(size=14),
                  legend.key.size = unit(1,"cm"),
                  axis.text.y = element_text(size = 14)) +
            ylab("") + xlab("TYK2 Amino Acid Position") +
            facet_grid(condition ~ lib)
```

```{r, fig.width=35, fig.height=35}
heat_l2fc
```
```{r, fig.width=35, fig.height=35}
heat_z
```
