---
title: FlowDMS Assay Report
output: github_document
---

```{r, echo=FALSE, warning=FALSE, message=FALSE}
knitr::opts_chunk$set(
  echo = FALSE,
  warning = FALSE,
  message = FALSE
)
```

## FlowDMS Assay Report

1. [Barcode Sequencing Distributions](#part1)
2. [Variant Effect Profiles and Midpoints](#part2)
3. [Visualizations](#part3)

```{r}
library(magrittr)
library(data.table)
library(ggrepel)
library(ggpubr)
library(vroom)
library(scales)
library(scico)
library(ggridges)
library(knitr)
library(ggbeeswarm)
library(fs)
library(patchwork)
library(ggnewscale)
library(furrr)
library(tidyverse)

source("/home/rstudio/dms/src/dms-analysis-utils.R")
```

### Barcode Sequencing Distributions <a name="part1"></a>

```{r}
mapped_counts <- read_tsv("/home/rstudio/dms/pipeline/tyk2-vamp-final-vampseq-split.mapped-counts.tsv")

bc_counts_aa <- mapped_counts %>%
    filter(mut_aa != "WT") %>%
    group_by(sample,wt_aa,mut_aa,pos) %>%
    count()

bc_counts_codon <- mapped_counts %>%
    filter(mut_aa != "WT") %>%
    group_by(sample,wt_codon,mut_codon,pos) %>%
    count()

res_count <- bc_counts_aa %>%
    ggplot() +
        geom_boxplot(aes(x = sample, y = n)) +
        theme_pubr() +
        geom_hline(yintercept = 30, color = "red") +
        ylab("number of unique barcodes") + ggtitle("AA-level unique barcode counts")

codon_count <- bc_counts_codon %>%
    ggplot() +
        geom_boxplot(aes(x = sample, y = n)) +
        theme_pubr() +
        geom_hline(yintercept = 30, color = "red")  +
        ylab("number of unique barcodes") + ggtitle("Codon-level unique barcode counts")

res_count_log <- bc_counts_aa %>%
    ggplot() +
        geom_boxplot(aes(x = sample, y = log10(n))) +
        theme_pubr() +
        geom_hline(yintercept = log10(30), color = "red") +
        ylab("number of unique barcodes") + ggtitle("AA-level unique barcode counts, log10")

codon_count_log <- bc_counts_codon %>%
    ggplot() +
        geom_boxplot(aes(x = sample, y = log10(n))) +
        theme_pubr() +
        geom_hline(yintercept = log10(30), color = "red")  +
        ylab("number of unique barcodes") + ggtitle("Codon-level unique barcode counts, log10")
```

```{r, fig.width=15, fig.height=6}
res_count + codon_count
res_count_log + codon_count_log 
```

To get a sense of positional distribution, we can show the same data as lineplots across the length of TYK2. Below is an example using the first sample; the remaining plots can be found [here](./coverage-plots):

```{r}
wt_tyk2 <- read_tsv("/home/rstudio/dms/data/tyk2/tyk2-wt-aa.tsv")
coverage_plots <- map(.x = unique(bc_counts_aa$sample),
                      .f = ~plot_coverage(bcs = bc_counts_aa,
                                          wt_aa = wt_tyk2,
                                          sample_id = .,
                                          destdir = "OCNT-VAMPLIB-1-assay-run3"))
```

```{r, fig.width=10, fig.height=20}
coverage_plots[[1]]
```

```{r, fig.width=12, fig.height=10}
mapped_counts %>%
    ungroup() %>%
    count(sample, chunk) %>%
    mutate(chunk = gsub("rc", "", chunk),
           chunk = as.numeric(chunk)) %>%
    ggplot() +
        geom_bar(aes(x = chunk, y = n), width = 1, color = NA, fill = "black", stat = "identity") +
        theme_bw() +
        facet_wrap(~sample, ncol = 4) +
        xlab("Chunk") + ylab("Total Number of Unique Barcodes\nAcross All Variants")
```

```{r, fig.width=10, fig.height=6}
mapped_counts %>%
    ungroup() %>%
    count(sample) %>%
    ggplot() +
        geom_bar(aes(x = sample, y = n), width = 1, color = NA, fill = "black", stat = "identity") +
        theme_bw(base_size = 20) +
        xlab("Sample") + ylab("Total Number of Unique Barcodes\nAcross All Variants")
```

### Variant Effect Profiles and Midpoints <a name="part2"></a>

```{r}
raw_sumstats <- read_tsv("/home/rstudio/bms-dms/sumstats/OCNT-VAMPLIB-1-assay-run3-vampseq.sumstats.tsv")
sumstats <-  raw_sumstats %>%
    filter(grepl("aa",term), pos != 1188) %>%
    separate(term, c("condition", "aa"), ":") %>%
    mutate(estimate = estimate / log(2),
           std.error = std.error / log(2),
           condition = gsub("condition_conc", "", condition),
           group = if_else(aa %in% c("mut_aa*", "mut_aaX"), chunk, "Non-Stop"),
           aa = gsub("mut_aa", "", aa),
           aa = if_else(aa %in% c("*","X"), "*", aa))

marginals <- raw_sumstats %>% filter(is.na(term))
weights <- marginals %>%
    dplyr::select(condition_conc, chunk, pos, mut_aa, estimate, std.error) %>%
    rename("bin" = "condition_conc") %>%
    group_by(chunk, pos, mut_aa) %>%
    mutate(bin = case_when(bin == 25 ~ 0.125,
                           bin == 50 ~ 0.375,
                           bin == 75 ~ 0.625,
                           TRUE ~ 0.875))

aa_uniq <- unique(sumstats$aa)
pos_uniq <- 1:1187

weights_nest <- weights %>% nest(data = c(-chunk, -pos, -mut_aa))
metadata_df <- weights_nest %>% dplyr::select(-data)
```

```{r}
# this cell takes a few minutes to run
plan(multicore, workers = 40)
midpoints <- future_map2_dfr(.x = weights_nest$data,
                             .y = transpose(metadata_df),
                             ~generate_resamples(.x$estimate,
                                                 .x$std.error,
                                                 .y,
                                                 num = 1000))
```

```{r}
wt_scores <- midpoints %>%
    filter(mut_aa == "WT") %>% 
    group_by(chunk) %>%
    summarize("WT score" = median(score_mean),
              "WT score standard error" = median(score_se))

wt_scores %>% kable("markdown")
```

```{r}
midpoints_test <- midpoints %>%
    filter(mut_aa != "WT") %>%
    left_join(wt_scores, by = "chunk") %>%
    mutate(estimate = score_mean - `WT score`,
           std.error = sqrt(score_se^2 + `WT score standard error`^2),
           statistic = estimate/std.error,
           p.value = pmin(pnorm(statistic, mean = 0, sd = 1)*2,
                                 (1-pnorm(statistic, sd = 1))*2),
           p.adj = p.adjust(p.value, method = "BH"))

midpoints_test %>%
    write_tsv("/home/rstudio/bms-dms/sumstats/TYK2-VAMP/OCNT-VAMPLIB-1-assay-run3-vampseq-midpoints.sumstats.tsv")
```

```{r, fig.width=15, fig.height=8}
midpoints %>%
    mutate(group = case_when(mut_aa %in% c("*", "X") ~ chunk,
                             mut_aa == "WT" ~ "WT",
                             TRUE ~ "Non-Stop"),
           colors = case_when(group == "Non-Stop" ~ "black",
                              group == "WT" ~ "blue",
                              TRUE ~ "red")) %>% 
    ggplot() +
        geom_density(aes(x = score_mean,
                         group = group,
                         color = colors,
                         size = colors)) +
        theme_bw(base_size = 15) +
        scale_color_identity() +
        scale_size_manual(values = c("blue" = 1, "black" = 2, "red" = 0.5)) +
        theme(legend.pos = "none") +
        xlab("Midpoint") + ylab("Density") +
        ggtitle("TYK2 Full Length FlowDMS: Midpoints")
```

```{r, fig.width=15, fig.height=8}
midpoints_test %>%
    mutate(group = case_when(mut_aa %in% c("*", "X") ~ chunk,
                             mut_aa == "WT" ~ "WT",
                             TRUE ~ "Non-Stop"),
           colors = case_when(group == "Non-Stop" ~ "black",
                              group == "WT" ~ "blue",
                              TRUE ~ "red")) %>% 
    ggplot() +
        geom_density(aes(x = statistic,
                         group = group,
                         color = colors,
                         size = colors)) +
        theme_bw(base_size = 15) +
        scale_color_identity() +
        scale_size_manual(values = c("blue" = 1, "black" = 2, "red" = 0.5)) +
        theme(legend.pos = "none") +
        xlab("Z-Statistic [Mutant minus WT Midpoint]") + ylab("Density") +
        ggtitle("TYK2 Full Length FlowDMS: Z-Statistics")
```

```{r, fig.width=15, fig.height=8}
midpoints_test %>%
    mutate(group = if_else(mut_aa %in% c("X","*"), "red", "black"),
           chunk = if_else(chunk == "12rc", "12", chunk)) %>%
    ggplot() +
        geom_density(aes(x = statistic,
                         group = group,
                         color = group)) +
        theme_bw(base_size = 15) +
        scale_color_identity() +
        theme(legend.pos = "none") +
        xlab("Z-Statistic [Mutant minus WT Midpoint]") + ylab("Density") +
        ggtitle("TYK2 Full Length FlowDMS: Z-Statistics") +
        facet_wrap(~as.numeric(chunk))
```

```{r, fig.width=20, fig.height=7}
midpoints %>%
    filter(mut_aa %in% c("X","*")) %>%
    ggplot() +
        geom_pointrange(aes(x = pos, y = score_mean,
                            ymin = score_mean - score_se*2,
                            ymax = score_mean + score_se*2), size = 0.5) +
        theme_pubr(base_size = 18) +
        ggtitle("TYK2 Full Length FlowDMS: Stop Midpoints") +
        xlab("TYK2 Position") + ylab("Midpoint +/- 2 Standard Errors")
```

```{r, fig.width=15, fig.height=5}
midpoints_test %>%
    mutate(mut_aa = if_else(mut_aa %in% c("*", "X"), "Stop", mut_aa)) %>%
    rename("midpoint shift" = "statistic") %>%
    ggplot() +
            geom_tile(aes(x = pos, y = fct_rev(mut_aa), fill = `midpoint shift`)) +
            scale_fill_scico_mid(palette = "vik", mid = 0, oob = squish, limits = c(-15, 5)) +
            theme_pubr(base_size = 16, x.text.angle = 45) +
            theme(legend.text=element_text(size=14),
                  legend.key.size = unit(1,"cm"),
                  axis.text.y = element_text(size = 14)) +
            ylab("") + xlab("TYK2 Amino Acid Position") +
            ggtitle("Unnormalized")
```