---
title: "`r paste(params$description, '-', params$exp_code)`"
author: "Robert Warneford-Thomson"
editor: visual
format: 
  gfm:
    toc: true
    number-sections: true
code-fold: true
execute: 
  warning: false
params:
  description: "TYK2 DMS manuscript - Fig 1 supplement"
  date: !expr format(Sys.time(), '%Y%m%d')
editor_options: 
  chunk_output_type: console
knitr:
  opts_chunk:
    fig.align: "center"
---

# I. SETUP

# Chunk options

```{r opts}
knitr::opts_chunk$set(
  echo = TRUE,
  root.dir = "~/Analyses/bms-dms/paper",
  fig.path = "./supplementary/",
  fig.dpi=600,
  warning = FALSE,
  message = FALSE, 
  dev = c("png", "pdf")
)
```

### Packages

```{r}
pacman::p_load(
  colorspace,
  ggbeeswarm,
  ggnewscale,
  ggh4x,
  ggpubr,
  ggrastr,
  ggsci,
  ggsignif,
  magrittr,
  readxl,
  paletteer,
  patchwork,
  scico,
  tidyverse
)

scale_fill_scico_mid <- function(..., mid = 0, alpha = NULL, begin = 0, end = 1, direction = 1, reverse = TRUE, palette = "broc") {
  force(mid)
  ggplot2::continuous_scale(
    aesthetics = "fill",
    scale_name = "gradient2",
    palette = scales::gradient_n_pal(
      colours = scico::scico(256, alpha, begin, end, direction, palette),
      values = NULL, space = "Lab"
    ),
    guide = "colourbar",
    rescaler = function(x, to = c(0, 1), from = range(x, na.rm = TRUE)) {
      scales::rescale_mid(x, to, from, mid)
    },
    ...
  )
}
```

### Variables

```{r setup}

cbPalette <- c("#DC5E65", "#E69F00", "#56B4E9", "#0072B2", "darkgrey", "#009E73", "#F0E442", "pink", "#CC79A7", "lightgrey", "grey")

aa_order <- c(
  "*", "P", "G", "A", "M", "V", "L", "I", "T", "S",
  "C", "Q", "N", "Y", "W", "F", "E", "D", "K", "H", "R"
)
```

# Functions

## compute_difference

```{r compute_difference}
compute_difference <- function(test, control, sumstats) {
  df1 <- sumstats %>%
    dplyr::filter(condition == test) %>%
    select(-condition)
  df2 <- sumstats %>%
    dplyr::filter(condition == control) %>%
    select(-condition)

  df <- inner_join(df1, df2,
    by = c("pos", "chunk", "aa", "version")
  )

  new_stats <- df %>%
    mutate(
      log2FoldChange = log2FoldChange.x - log2FoldChange.y,
      log2StdError = sqrt(log2StdError.x^2 + log2StdError.y^2)
    ) %>%
    select(pos, chunk, aa, log2FoldChange, log2StdError, version) %>%
    ungroup()

  new_stats$condition <- paste0(test, " - ", control)

  return(new_stats)
}
```

## theme_pub

```{r}
theme_pub <- function(base_size = 11, base_family = "") {
  # based on https://github.com/noamross/noamtools/blob/master/R/theme_nr.R
  # start with theme_bw and modify from there!
  theme_bw(base_size = base_size, base_family = base_family) + # %+replace%
    theme(
      # grid lines
      panel.grid.major.x = element_line(colour = "#ECECEC", linewidth = 0.5, linetype = 1),
      panel.grid.minor.x = element_blank(),
      panel.grid.minor.y = element_blank(),
      panel.grid.major.y = element_line(colour = "#ECECEC", linewidth = 0.5, linetype = 1),
      panel.background = element_blank(),

      # axis options
      # axis.ticks.y = element_blank(),
      axis.title.x = element_text(size = rel(1), vjust = 0.25),
      axis.title.y = element_text(size = rel(1), vjust = 0.35),
      # axis.text      = element_text(color="black", size=rel(1)),
      axis.text.y = element_text(size = rel(1), color = "black"),
      axis.text.x = element_text(angle = 45, color = "black", hjust = 1, size = rel(1)),

      # legend options
      legend.title = element_text(size = rel(1)),
      legend.key = element_rect(fill = "white"),
      legend.key.size = unit(0.02, "npc"),
      legend.text = element_text(size = rel(1)),

      # facet options
      strip.text = element_text(size = rel(1)),
      strip.background = element_blank(),

      # title options
      plot.title = element_text(size = rel(1), hjust = 0.5),
      plot.subtitle = element_text(size = rel(0.8), hjust = 0),
    )
}
```

# Load and format data

After discussing with Conor, will use 1 and 10 U/mL IFNa data from run3, and 100 U/mL IFNa from run7

```{r load-data}
# load IFNa sumstats
ifna_sumstats_run3 <- read_tsv("../sumstats/archive/TYK2-run3-combined-cleaned.sumstats.tsv", show_col_types = FALSE) %>%
  # dplyr::filter(condition == "IFNalpha100_0") %>%
  mutate(
    aa = if_else(aa %in% c("X", "Stop", "*"), "*", aa),
    aa = factor(aa, levels = aa_order)
  )

# compute contrasts for IFNa data
conditions <- unique(ifna_sumstats_run3$condition)
noneIdx <- which(conditions == "None_0")

ifna_run3 <- map(
  conditions[-noneIdx],
  ~ compute_difference(
    test = .,
    control = "None_0",
    sumstats = ifna_sumstats_run3
  )
) %>%
  list_rbind() %>%
  mutate(condition = factor(condition,
    levels = c(
      "IFNbeta_100 - None_0",
      "IFNalpha_1 - None_0",
      "IFNalpha_10 - None_0",
      "IFNalpha_100 - None_0",
      "IFNalphaWithDrug_100 - None_0"
    )
  )) %>%
  dplyr::filter(condition %in% c("IFNalpha_1 - None_0", "IFNalpha_10 - None_0", "IFNalpha_100 - None_0")) %>%
  mutate(
    z_statistic = log2FoldChange / log2StdError,
    p.value = pmin(
      pnorm(z_statistic, mean = 0, sd = 1) * 2,
      (1 - pnorm(z_statistic, sd = 1)) * 2
    )
  ) %>%
  mutate(run = "run3")

# load run7 IFN100 data
ifna_sumstats_run7 <- read_tsv("../sumstats/archive/TYK2-run7-combined-cleaned.sumstats.tsv", show_col_types = FALSE) %>%
  # dplyr::filter(condition == "IFNalpha100_0") %>%
  mutate(
    aa = if_else(aa %in% c("X", "Stop", "*"), "*", aa),
    aa = factor(aa, levels = aa_order)
  )

# compute contrasts for IFNa data
conditions <- unique(ifna_sumstats_run7$condition)
noneIdx <- which(conditions == "None_0")

ifna_run7 <- map(
  conditions[-noneIdx],
  ~ compute_difference(
    test = .,
    control = "None_0",
    sumstats = ifna_sumstats_run7
  )
) %>%
  list_rbind() %>%
  mutate(condition = factor(condition,
    levels = c(
      "IFNalpha100_0 - None_0",
      "IFNalpha100+BMS-986202_2e-08 - None_0",
      "IFNalpha100+Ropsacitinib_1e-05 - None_0",
      "IFNalpha100+Zasocitinib_7e-09 - None_0"
    )
  )) %>%
  dplyr::filter(condition == "IFNalpha_100 - None_0") %>%
  mutate(
    z_statistic = log2FoldChange / log2StdError,
    p.value = pmin(
      pnorm(z_statistic, mean = 0, sd = 1) * 2,
      (1 - pnorm(z_statistic, sd = 1)) * 2
    )
  ) %>%
  mutate(run = "run7")

# merge experiments
ifna_merge <- bind_rows(ifna_run3, ifna_run7) %>%
  mutate(run = factor(run, levels = c("run7", "run3"))) %>% 
  arrange(run) %>% 
  # for IFNalpha100_0, keep sumstats from run7 only where possible, default to run3 when not found in run7
  group_by(pos, aa, condition) %>% 
  slice_head(n = 1) %>%
  ungroup() %>% 
  pivot_longer(c("log2FoldChange"), names_to = "param", values_to = "param_estimate") %>%
  select(pos, aa, chunk, param, run, param_estimate, param_se = log2StdError, condition, z_statistic, p.value) %>%
  mutate(
    assay = "IFNa",
    p.adj = p.adjust(p.value, method = "BH")
  )

stability <- read_tsv("../sumstats/archive/TYK2-FLOW-flow-cleaned.midpoints.tsv", show_col_types = FALSE) %>%
  mutate(
    aa = if_else(aa %in% c("X", "Stop", "*"), "*", aa),
    aa = factor(aa, levels = aa_order)
  ) %>%
  pivot_longer(c("midpoint_shift"), names_to = "param", values_to = "param_estimate") %>%
  select(pos, aa, chunk, param, param_estimate, param_se = midpoint_shift_se, z_statistic = statistic, p.value, p.adj) %>%
  mutate(assay = "Flow")

merge_sumstats <- bind_rows(ifna_merge, stability) %>%
  # add annotation labels for plotting
  mutate(facet_label = case_when(
    assay == "Flow" ~ "Abundance",
    assay == "IFNa" ~ paste0(
      "IFN\u03B1 ", str_extract(condition,
        pattern = "IFNalpha_?(\\d+)",
        group = 1
      ),
      " U/mL"
    )
  )) %>%
  mutate(facet_label = fct_relevel(facet_label, "Abundance", after = Inf))

#annotate data with 3 letter aa
aalet <- read_tsv("data/aa-letters.tsv", col_names = c("full", "thr", "aa")) %>%
    select(-full) 

wtseq <- read_tsv("data/TYK2-sequence.tsv", col_names = c("pos", "thr"), skip = 1) %>%
    left_join(aalet, by = "thr") %>%
    rename("wt_aa" = "aa",
           "wt_thr" = "thr")
  
merge_sumstats <- merge_sumstats %>%
  left_join(wtseq, by = "pos") %>% 
  left_join(aalet %>% rename("mut_thr" = "thr"), by = c("aa")) %>% 
  mutate(aa = factor(aa, levels = aa_order)) %>%
  replace_na(replace = list(mut_thr = "Stop")) %>% 
  mutate(mut = paste0(wt_thr, pos, mut_thr)) %>% 
  #remove amino acid 1188
  dplyr::filter(pos != 1188)
```

# Load metadata

```{r annotate-data}
#load variant annotations / scores
tyk2_clinvar <- read_xlsx("data/clinvar_TYK2_20250109.xlsx")  %>% 
    select(Name,
           gene = `Gene(s)`,
           clinvar_class = `Germline classification`,
           mut = `Protein Change (header)`) %>% 
  mutate(clinvar_class = gsub("\\(.*","",clinvar_class)) %>%
  #remove Gly512Arg variant with Uncertain significance
  dplyr::filter(!(mut == "Gly512Arg" & clinvar_class == "Uncertain significance")) %>% 
  #aggregate benign and likely benign into single category
  mutate(clinvar_class = case_when(clinvar_class %in% c("Benign", "Benign/Likely benign", "Likely benign") ~ "Benign/Likely benign",
                                   .default = clinvar_class)) %>% 
  distinct()

eve_scores_b05 <- read_csv("data/TYK2_HUMAN_b05_20000_samples.csv") %>%
    mutate(mutations = gsub("^.","",mutations)) %>%
    select(-protein_name) %>%
    dplyr::filter(mutations != "t") %>%
    separate(mutations, c("pos", "aa"), sep = -1) %>%
    rename("EVE Score 1" = "evol_indices") %>%
    mutate(pos = as.numeric(pos))

eve_scores_b03 <- read_csv("data/TYK2_HUMAN_b03_20000_samples.csv") %>%
    mutate(mutations = gsub("^.","",mutations)) %>%
    select(-protein_name) %>%
    dplyr::filter(mutations != "t") %>%
    separate(mutations, c("pos", "aa"), sep = -1) %>%
    rename("EVE Score 2" = "evol_indices") %>%
    mutate(pos = as.numeric(pos))

eve_scores <- inner_join(eve_scores_b05, eve_scores_b03, by = c("pos", "aa"))

tyk2_sift <- read_delim("data/TYK2-sift.tsv.gz", delim = " ") %>%
    pivot_longer(names_to = "aa", values_to = "sift", A:Y) %>%
    mutate(mut_sift = paste0(wt, aa)) %>%
    select(mut_sift, sift)

tyk2_polyp <- read_tsv("data/TYK2-polyphen2.tsv.gz") %>%
    select(o_pos, o_aa2, pph2_prob) %>%
    rename("PolyPhen2" = "pph2_prob")

alphamissense <- read_tsv("data/TYK2.Uniprot.P29597.AlphaMissense_aa.substitutions.tsv",
                          col_names = c("uniprot_id" ,"mut_id", "alphamissense_score", "AlphaMissense Class")) 

#annotate sumstats
anno_sumstats <- merge_sumstats %>%
  left_join(tyk2_clinvar, by = "mut", relationship = "many-to-many") %>% 
  replace_na(replace = list(clinvar_class = "not in ClinVar")) %>% 
  mutate(mut_sift = paste0(pos, wt_aa, aa),
         mut_alphaM = paste0(wt_aa, pos, aa)) %>%
  left_join(tyk2_sift, by = "mut_sift") %>% 
  left_join(tyk2_polyp, by = c("pos" = "o_pos",
                                 "aa" = "o_aa2")) %>%
  left_join(alphamissense, by = c("mut_alphaM" = "mut_id")) %>% 
  select(!c(mut_sift, mut_alphaM))

anno_df <- anno_sumstats %>% 
  dplyr::filter(mut_thr != "Stop") %>% 
  dplyr::filter(assay == "Flow" | run == "run7") %>%
  pivot_longer(c("clinvar_class", "AlphaMissense Class"),
               names_pattern = "(clinvar|AlphaMissense).*",
               names_to = "classifier",
               values_to = "classifier_value") %>%
  mutate(classifier_value = str_to_title(classifier_value)) %>% 
  mutate(classifier_value = factor(classifier_value, levels = c("Pathogenic",
                                                               "Conflicting Classifications Of Pathogenicity",
                                                               "Uncertain Significance",
                                                               "Ambiguous",
                                                               "Likely Benign",
                                                               "Benign/Likely Benign",
                                                               "Benign",
                                                               "Not In Clinvar")))
```

# Figures

## Fig S1x1 - Flow + IFNa-1,10,100 heatmaps

```{r fig-S1x1, fig.width=8.5, fig.height=11}
S1x1 <- merge_sumstats %>%
  mutate(facet_label = str_replace_all(facet_label, "([0-1]+)", "\n\\1")) %>% 
  mutate(facet_label = fct_relevel(facet_label, "Abundance", after = Inf)) %>% 
  mutate(aa = fct_recode(aa, "Stop" = "*")) %>% 
  ggplot() +
  facet_grid2(
    rows = "facet_label",
    drop = FALSE,
    scales = "fixed",
    axes = "all"
  ) + 
  geom_tile(aes(x = pos, y = aa, fill = z_statistic)) +
  scale_fill_scico_mid("Z statistic", limits = c(-10, 2.5),
                       palette = "vik", mid = 0,
                       oob = scales::squish) +
  scale_x_continuous(
    breaks = seq(0, 1000, by = 250),
    expand = c(0, 0)
  ) +
  ylab("") +
  xlab("TYK2 Amino Acid Position") +
  theme_pub(base_size = 9) +
  theme(
    axis.text.x = element_text(angle = 0, hjust = 0.5,
                               size = unit(8, "points")),
    legend.key.width = unit(0.05, "npc"),
    axis.text.y = element_text(size = unit(8, "points")),
    legend.position = "top",
    legend.text = element_text(size = unit(8, "points")),
    legend.title = element_text(margin = margin(0,10,0,0)),
    legend.box.spacing = unit(0.001, "npc"),
    strip.text.y = element_text(hjust = 0.5, angle = 0, size = unit(8, "points")),
  ) 
S1x1
```

## Fig S1x2

### Fig S1x2-A - Barplots with LoF/GoF counts

I'm going to just use a 1% FDR to determine GoF and LoF variants with no minimum log2fold or midpoint shift threshold

```{r S1x2-A, fig.width=5, fig.height=5}
A <- merge_sumstats %>%
  # dplyr::filter(assay == "Flow" |) %>%
  mutate(facet_label = str_replace_all(facet_label, "([0-1]+)", "\n\\1")) %>% 
  mutate(facet_label = fct_relevel(facet_label, "Abundance", after = Inf)) %>%
  mutate(status = case_when(param_estimate < 0 & p.adj < 0.01 ~ "LoF",
    param_estimate > 0 & p.adj < 0.01 ~ "GoF",
    .default = "no effect"
  )) %>%
  mutate(status = factor(status, levels = c("GoF", "no effect", "LoF"))) %>%
  ggplot(aes(x = facet_label, fill = status)) +
  geom_bar(position = "fill") +
  geom_label(aes(label = after_stat(count)),
    stat = "count",
    position = position_fill(vjust = 0.5),
    show.legend = FALSE,
    size = rel(2),
  ) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual("", values = c("#DC5E65", "lightgrey", "#0072B2")) +
  theme_pub() +
  labs(
    
  ) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        legend.position = "top",
        panel.grid.major.x = element_blank(),
        axis.title.x  = element_blank(),
        axis.title.y  = element_blank(),
        legend.margin=margin(0,0,-25,0),
        legend.box.margin=margin(-10,-10,-10,-10),
        legend.text = element_text(size = rel(0.5)),
        legend.key.size = unit(0.01, "npc"),
        axis.text= element_text(size = unit(6, "points")),
        plot.tag.position = c(0.01, 1.01)
        )
```

### Fig S1x2-B - stop effect density plots

```{r S1x2-B, fig.width=8, fig.height=8}
B <- merge_sumstats %>%
  # dplyr::filter(grepl("- None0", condition)) %>%
  mutate(group = factor(if_else(aa == "*", "stop variants", "other variants"),
                        levels = c("stop variants", "other variants"))) %>%
  ggplot() +
  geom_density(
    aes(
      x = z_statistic, group = group,
      fill = group
    ),
    color = NA,
    alpha = 0.5
  ) +
  geom_text(aes(label = facet_label), x = -15, y = 0.4, size = rel(2), hjust = 0) +
  facet_grid2(facet_label ~ ., scales = "fixed") +
  scale_fill_manual("", values = c(
    "stop variants" = "red",
    "other variants" = "black")) +
  labs(
    x = "Z-Statistic",
    y = "Density",
    # title = "Fig S1D",
    # subtitle = "IFN\u03B1 data normalized 0 cytokine condition"
  ) +
  coord_cartesian(xlim = c(-15, 7)) +
  scale_y_continuous(breaks = 0.5) +
  # scale_x_continuous(expand = c(0, 0)) +
  theme_pub(base_size = 12) +
  theme(
    axis.title = element_text(size = rel(0.6)),
    axis.text.x = element_text(angle = 0, hjust = 0.5),
    panel.grid.major.y = element_blank(),
    panel.grid.major.x = element_blank(),
    legend.box = element_blank(),
    legend.position = "top",
    legend.key = element_rect(color = NA),
    legend.margin = margin(0,0,-25,0),
    legend.box.spacing = unit(0.001, "npc"),
    legend.title = element_blank(),
    legend.text = element_text(size = rel(0.5)),
    legend.key.height = unit(0.01, "npc"),
    strip.text.y = element_blank(),# element_text(angle = 0, size = rel(0.7)),
    axis.text= element_text(size = unit(6, "points")),
    plot.tag.position = c(0.01, 0.9)
  )
```

### Fig S1x2-C - AlphaMissense

```{r S1x2-C, fig.width=8, fig.height=8}
am_df <-  anno_df %>% 
  dplyr::filter(classifier == "AlphaMissense")

C <- am_df %>%
    ggplot(aes(x = classifier_value, y = z_statistic,
             color = classifier_value)) +
  geom_quasirandom_rast(size = 0.5,
                        show.legend = FALSE) +
  # geom_violin(
  #   aes(fill = classifier_value),
  #   scale = "width",
  #   show.legend = FALSE
  #   ) +
  geom_boxplot(show.legend = FALSE,
               col = "black",
               outliers = FALSE,
               width = 0.09) +
  scale_color_manual(values = c(
    "Pathogenic" = "#DC5E65",
      "Ambiguous" =  "#E69F00",
      "Benign" =  "#0072B2")
  ) +
    scale_fill_manual(values = c(
    "Pathogenic" = "#DC5E65",
      "Ambiguous" =  "#E69F00",
      "Benign" =  "#0072B2")
  ) +
  # add significance calculation using KS test
  geom_signif(
    color = "black",
    comparisons = list(
      c("Pathogenic", "Ambiguous"),
      c("Pathogenic", "Benign"),
      c("Ambiguous", "Benign")
    ),
    map_signif_level = function(p) {
      if(p < 0.001) return("***")
      else if(p < 0.01) return("**")
      else if(p < 0.05) return("*")
      else return("ns")
    },
    test = "ks.test",
    margin_top = 0.01,
    step_increase = 0.025,
    tip_length = 0.005,
    vjust = 0.65,
    size = 0.5,
    textsize = 3
  ) +
  facet_grid2(. ~ facet_label,
              scales = "free_y",
              space = "free") +
  labs(
    x = "AlphaMissense annotation",
    y = "Z statistic",
  ) +
  ylim(-15,7) +
  theme_pub(base_size = 11) +
  theme(
    plot.title = element_blank(),
    axis.text.x = element_text(size = rel(0.8)),
    panel.grid.major.x = element_blank(),
    axis.title = element_text(size = rel(0.6)),
    strip.text.x = element_text(size = rel(0.7)),
    plot.tag.position = c(0.01, 0.98)
    )

```

### Fig S1x2-D - Fiducial variant forest plot

```{r S1x2-D, fig.width=12, fig.height=6}
spikeins <- read_tsv("../paper/data/tyk2-spikeins.tsv",
  col_names = c("pos", "mut_id", "class")
) %>%
  mutate(
    class = if_else(class == "-", "Unknown", class),
    class = factor(class, levels = c("LOF", "Unknown", "GOF"))
  ) %>%
  arrange(class, pos) %>%
  separate_wider_regex(mut_id,
    patterns = c(".*\\d+", "aa" = ".*"),
    cols_remove = FALSE
  ) %>%
  mutate(aa = ifelse(aa == "Stop", "*", aa)) 


spikeins$mut_id <- factor(spikeins$mut_id, levels = spikeins$mut_id)

spikeins_join <- spikeins %>%
  inner_join(merge_sumstats, by = c("pos", "aa")) %>%
  arrange(class, pos) %>%
  mutate(class = factor(class, levels = c("GOF", "Unknown", "LOF"))) %>%
  mutate(facet_label = str_replace_all(facet_label, "([0-1]+)", "\n\\1")) %>%
  mutate(facet_label = fct_relevel(facet_label, "Abundance", after = Inf)) %>% 
  mutate(significance = factor(if_else(p.adj < 0.01, "FDR < 1%", "not significant")))

D1 <- spikeins_join %>% 
  filter(facet_label != "Abundance") %>% 
  ggplot() +
  geom_vline(xintercept = 0) +
  geom_hline(yintercept = 26.5,
             color = "#DC5E65",
             linetype = "dashed") +
  geom_text(
    data = tibble(facet_label = "IFNα \n1 U/mL",
                  label = "GOF"),
    aes(label = label),
    color = "#DC5E65",
    y = 28.5,
    x = -1.5
  ) +
  geom_hline(yintercept = 15.5,
             color = "#0072B2",
             linetype = "dashed") +
    geom_text(
    data = tibble(facet_label = "IFNα \n1 U/mL",
                  label = "LOF"),
    aes(label = label),
    color = "#0072B2",
    y = 12.5,
    x = -1.5
  ) +
  geom_pointrange(
    aes(
      x = param_estimate,
      y = factor(mut_id),
      xmin = param_estimate - 2 * param_se,
      xmax = param_estimate + 2 * param_se,
      color = significance
    ),
    linewidth = 0.25,
    size = 0.1
  ) +
  theme_pubr(base_size = 13) +
  labs(
    x = expression(Log[2]~fold~change~"\u00B1"~2~standard~errors),
    y = "") +
    #," ± 2 standard errors"),
  #  subtitle = "IFNa data normalized to 0 cytokine control\nEffect size = log2FoldChange for IFNa data\nEffect size = midpoint shift for Abundance DMS (see methods)\nPoints in black (padj < 0.01)"
  scale_color_manual("", values = c("black", "lightgrey")) +
  xlim(c(-2.5, 1)) +
  theme_pub(base_size = 11) +
  theme(
    panel.grid.major.x = element_blank(),
    strip.text.x = element_text(size = unit(6, "points")),
    axis.title = element_text(size = rel(0.6)),
    legend.position = "top",
    legend.margin=margin(0,0,-10,0),
    axis.text.y = element_text(size = unit(6, "points")),
    axis.text.x = element_text(angle = 0, hjust = 0.6,
                               size = unit(6, "points")),
    panel.grid.major.y = element_blank(),
    plot.tag.position = c(0.01, 0.98)
  ) +
  facet_grid2(. ~ facet_label, scales = "fixed")

#panel for abundance DMS
D2 <- spikeins_join %>% 
  filter(facet_label == "Abundance") %>% 
  ggplot() +
  geom_vline(xintercept = 0) +
   geom_hline(yintercept = 26.5,
             color = "#DC5E65",
             linetype = "dashed") +
  geom_hline(yintercept = 15.5,
             color = "#0072B2",
             linetype = "dashed") +
  geom_pointrange(
    aes(
      x = param_estimate,
      y = mut_id,
      xmin = param_estimate - 2 * param_se,
      xmax = param_estimate + 2 * param_se,
      color = significance
    ),
    linewidth = 0.25,
    size = 0.1,
    show.legend = FALSE
  ) +
  theme_pubr(base_size = 13) +
  labs(
    x = "Midpoint shift\n± 2 standard errors",
    y = "",
    # title = "Fig S1E",
  #  subtitle = "IFNa data normalized to 0 cytokine control\nEffect size = log2FoldChange for IFNa data\nEffect size = midpoint shift for Abundance DMS (see methods)\nPoints in black (padj < 0.01)"
  ) +
  scale_color_manual("", values = c("black", "lightgrey")) +
  scale_x_continuous(limits = c(-1,1),
                     breaks = c(-1,0,1)) +
  theme_pub(base_size = 11) +
  theme(
    panel.grid.major.x = element_blank(),
    strip.text.x = element_text(size = unit(6, "points")),
    axis.title = element_text(size = rel(0.6)),
    axis.text.y = element_blank(),
    plot.margin = margin(0,0,0,10),
    axis.ticks.y = element_blank(),
    # axis.text.y = element_text(size = unit(4, "points")),
    axis.text.x = element_text(angle = 0, hjust = 0.6,
                               size = unit(6, "points")),
    panel.grid.major.y = element_blank(),
    plot.tag.position = c(0.01, 0.98)
  ) +
  facet_grid2(. ~ facet_label)

D <- D1 + (D2 + plot_layout(tag_level = "new")) +
  plot_layout(widths = c(3,1))
```

### Fig S1x2-E - ClinVar annotation

```{r S1x2-E, fig.width=8, fig.height=8}
anno_df <- anno_df %>% 
  mutate(classifier_value = fct_recode(classifier_value, "Conflicting\nClassifications\nOf Pathogenicity" = "Conflicting Classifications Of Pathogenicity",
                                       "Uncertain\nSignificance"  = "Uncertain Significance",
                                       "Benign/\nLikely Benign" = "Benign/Likely Benign"))

counts <- anno_df %>% 
  dplyr::filter(classifier == "clinvar") %>% 
  select(mut, classifier_value) %>% 
  distinct() %>% 
  group_by(classifier_value) %>% 
  summarise(count = n())

E <- anno_df %>% 
  dplyr::filter(classifier == "clinvar") %>% 
  left_join(counts) %>% 
  arrange(classifier_value) %>%
  # mutate(classifier_value_count = paste("count", count)) %>% 
  mutate(classifier_value_count =  fct_inorder(paste0(classifier_value, " (", format(count, big.mark = ",", trim = TRUE),")"))) %>%
  ggplot(aes(
    x = z_statistic,
    y = classifier_value_count,
    color = classifier_value_count,
             fill = classifier_value_count)) +
  geom_vline(xintercept = 0, color = "lightgrey", width = 0.5) +
  geom_quasirandom_rast(orientation = "y",
                   size = 0.5,
                   width = 0.2,
                   varwidth = TRUE,
                   show.legend = FALSE) +
  scale_color_manual(values = cbPalette) +
  scale_fill_manual(values = cbPalette) +
  facet_grid2(. ~ facet_label,
              scales = "free_x",
              space = "free") +
  labs(
    x = "Z statistic",
    y = "Clinvar annotation",
  ) +
  theme_pub(base_size = 11) + 
  theme(axis.text.x = element_text(angle = 0, hjust = 0.6),
        axis.text.y = element_text(angle = 0, hjust = 0.5),
        axis.text= element_text(size = unit(6, "points")),
        strip.text.y = element_text(angle = 0, size = unit(8, "points")),
        axis.title = element_text(size = unit(10, "points")),
        plot.tag.position = c(0.01, 0.98)
        )
```

## Merged Figure S1x2

```{r fig-S1x2-merged, fig.width=8.5, fig.height=11}
layout <- c(
  area(1, 1, 1, 1),
  area(1, 2, 1, 2),
  area(1, 3, 1, 3),
  area(2, 1, 2, 2),
  area(2, 3, 2, 3)
)
A + free(B, type = "label") + free(C, side = "l") + free(D) + E +
  plot_layout(heights = c(1, 1),
              guides = "keep",
              design = layout) +
  plot_annotation(tag_levels = "A") &
  theme(
    legend.key = element_blank(),
    panel.grid.major.y = element_blank(),
    axis.title = element_text(size = unit(8, "points")),
    axis.text = element_text(size = unit(6, "points")),
    plot.tag = element_text(size = unit(14, "points")),
    # plot.title = element_text(size = unit(14, "points")),
    legend.text = element_text(size = unit(8, "points")),
    # strip.text.x = element_blank(),
    strip.text.x = element_text(size = unit(6, "points")),
    plot.tag.position = c(0.01, 1.01)
  )

```
